WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -1187,23 +1154,111 @@ protected Boolean doInBackground(Void... params) {
 
         @Override
         protected void onPostExecute(Boolean saved) {
-            if (saved) {
-                Intent i = new Intent();
-                i.putExtra(EXTRA_SAVED_AS_LOCAL_DRAFT, true);
-                i.putExtra(EXTRA_IS_PAGE, mIsPage);
-                setResult(RESULT_OK, i);
-                ToastUtils.showToast(EditPostActivity.this, R.string.editor_toast_changes_saved);
+            saveResult(saved, true);
+            finish();
+        }
+    }
+
+    private void saveResult(boolean saved, boolean savedLocally) {
+        Intent i = getIntent();
+        i.putExtra(EXTRA_SAVED_AS_LOCAL_DRAFT, savedLocally);
+        i.putExtra(EXTRA_IS_PAGE, mIsPage);
+        i.putExtra(EXTRA_HAS_CHANGES, saved);
+        i.putExtra(EXTRA_POST, mPost);
+        setResult(RESULT_OK, i);
+    }
+
+    private void publishPost() {
+
+        boolean isFirstTimePublish = isFirstTimePublish();
+
+        boolean postUpdateSuccessful = updatePostObject();
+        if (!postUpdateSuccessful) {
+            return;
+        }
+
+        boolean isPublishable = PostUtils.isPublishable(mPost);
+
+        // if post was modified or has unsaved local changes and is publishable, save it
+        saveResult(isPublishable, false);
+
+        if (isPublishable) {
+            if (NetworkUtils.isNetworkAvailable(this)) {
+                if (!hasFailedMedia()) {
+                    savePostOnlineAndFinishAsync(isFirstTimePublish);
+                }
+            } else {
+                savePostLocallyAndFinishAsync();
+            }
+        } else {
+            ToastUtils.showToast(EditPostActivity.this, R.string.error_publish_empty_post, Duration.SHORT);
+        }
+    }
+
+    private void savePostAndFinish() {
+
+        // check if the opened post had some unsaved local changes
+        boolean hasLocalChanges = mPost.isLocallyChanged() || mPost.isLocalDraft();
+        boolean isFirstTimePublish = isFirstTimePublish();
+
+        boolean postUpdateSuccessful = updatePostObject();
+        if (!postUpdateSuccessful) {
+            return;
```

Comment:
It seems that currently, `updatePostObject()` can fail only when the `EditorFragment.java` fragment is not added to the activity. I think it's OK to silently return in this case.

Since this is not exactly obvious by looking at the code here, may I suggest adding a small comment about it? Something along the lines of "just return since the only case updatePostObject() can fail is when the editor fragment not being added to the activity". WDYT?